pkg/endpoint/connector/add.go

```diff
@@ -37,7 +38,8 @@ const (
 
 // Endpoint2IfName returns the host interface name for the given endpointID.
 func Endpoint2IfName(endpointID string) string {
-	return hostInterfacePrefix + truncateString(endpointID, 5)
+	sum := fmt.Sprintf("%x", sha256.Sum256([]byte(endpointID)))
```

Comment:
@nirmoy Won't this end up having the same problem? The first 5 characters of the sha256 can colide. What about using the hexadecimal format of the endpoint ID without being truncated?

Reply:
@aanm len(namespace) can be > len(ifname). 
chapman:~ # ip netns add 1234567890123456783
chapman:~ # ip link add 123456789012346 type dummy 
chapman:~ # ip link add 1234567890123467 type dummy 
Error: argument "1234567890123467" is wrong: "dev" not a valid ifname

Reply:
@nirmoy I thought the endpointID of the function was the endpoint ID and not the container ID. I was suggestion we could simply use the endpoint ID for the Ifname

"lxc${cilium_endpoint_id_in_hex}"

Reply:
updated..